crates/nostr/src/nips/nip62.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2022-2023 Yuki Kishimoto
+// Copyright (c) 2023-2025 Rust Nostr Developers
+// Distributed under the MIT software license
+
+//! NIP-62: Request to Vanish
+//!
+//! https://github.com/nostr-protocol/nips/blob/master/62.md
+
+use alloc::vec::Vec;
+
+use crate::RelayUrl;
+
+/// Request to Vanish target
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum VanishTarget {
+    /// Request to vanish from all relays
+    AllRelays,
+    /// Request to vanish from a specific list of relays.
+    Relays(Vec<RelayUrl>),
+}
+
+impl VanishTarget {
+    /// Vanish from a single relay
+    #[inline]
+    pub fn relay(relay: RelayUrl) -> Self {
+        Self::Relays(vec![relay])
+    }
+
+    /// Vanish from multiple relays
+    #[inline]
+    pub fn relays<I>(relays: I) -> Self
+    where
+        I: IntoIterator<Item = RelayUrl>,
+    {
+        Self::Relays(relays.into_iter().collect())
+    }
```

Comment:
@TheAwiteb, here I've removed the is empty logic check. Since we are using the enum, if a user want to erase the data from all relays must explicitly use `VanishTarget::all_relays()`. This to avoid to potentially send a global vanish request if the iterator is accidentally empty.

Reply:
> This to avoid to potentially send a global vanish request if the iterator is accidentally empty.

But this will produce an invalid event, and as I read before on an issue, that the crate should not produce invalid events

Reply:
The check is done in the `EventBuilder::request_vanish_with_reason`. And, also if the check is done in `VanishTarget::relays`, a dev can always do this, skipping the check: `VanishTarget::Relays(Vec::new())`

Reply:
https://github.com/rust-nostr/nostr/blob/c367ef1ea7cd451844c6115109453b7ce1cf02b6/crates/nostr/src/event/builder.rs#L742